File: solutionFiles/ReverseInteger.py
import json
import sys

def reverse_integer(x):
    # Check for the sign of the number
    sign = -1 if x < 0 else 1
    x *= sign

    # Reverse the digits of the number
    reversed_x = int(str(x)[::-1])

    # Restore the sign
    reversed_x *= sign

    # Check if the reversed integer is within the 32-bit signed integer range
    if reversed_x < -2**31 or reversed_x > 2**31 - 1:
        return 0

    return reversed_x

def main():
    # Read the input JSON string from the program arguments
    input_json = sys.argv[1]

    # Parse the JSON string to get the integer
    input_data = json.loads(input_json)
    x = input_data["x"]

    # Get the reversed integer
    result = reverse_integer(x)

    # Prepare the output as a JSON string
    output_json = json.dumps({"result": result})

    # Print the output JSON string
    print(output_json)

if __name__ == "__main__":
    main()

File: solutionFiles/ReverseInteger.java
public class ReverseInteger {
    public static void main(String[] args) {
        if (args.length != 1) {
            System.out.println("Usage: java ReverseInteger '{\"x\": 123}'");
            return;
        }

        // Read the input JSON string from the program arguments
        String inputJson = args[0];

        // Parse the JSON string to get the integer
        int x = parseJson(inputJson);

        // Get the reversed integer
        int result = reverseInteger(x);

        // Prepare the output as a JSON string
        String outputJson = "{\"result\": " + result + "}";

        // Print the output JSON string
        System.out.println(outputJson);
    }

    private static int parseJson(String json) {
        // Find the index of the colon to locate the start of the number
        int colonIndex = json.indexOf(':');
        // Find the index of the closing curly brace
        int endIndex = json.indexOf('}', colonIndex);

        // Extract the substring representing the number and trim any whitespace
        String numberStr = json.substring(colonIndex + 1, endIndex).trim();

        // Convert the substring to an integer
        return Integer.parseInt(numberStr);
    }

    public static int reverseInteger(int x) {
        // Check for the sign of the number
        int sign = x < 0 ? -1 : 1;
        x = Math.abs(x);

        // Reverse the digits of the number
        int reversed = 0;
        while (x != 0) {
            int pop = x % 10;
            x /= 10;
            // Check for overflow
            if (reversed > (Integer.MAX_VALUE - pop) / 10) {
                return 0;
            }
            reversed = reversed * 10 + pop;
        }

        // Restore the sign
        reversed *= sign;

        return reversed;
    }
}
